docs/lang_spec/Cadence Language Specification.md

```diff
@@ -0,0 +1,5798 @@
+  - Strings support escape sequences for Unicode scalars and special characters.
+    - `\0`: Null character
+    - `\\`: Backslash
+    - `\t`: Horizontal tab
+    - `\n`: Line feed
+    - `\r`: Carriage return
+    - `\"`: Double quotation mark
+    - `\'`: Single quotation mark
+    - `\u`: A Unicode scalar value, written as `\u{x}`, where `x` is a 1–8 digit hexadecimal number which needs to be a valid Unicode scalar value, i.e., in the range 0 to 0xD7FF and 0xE000 to 0x10FFFF inclusive
+    - Example: `"Hello, Cadence\u{21}"`, `"Multiline\nString"`.
+
+- **Boolean Literals**:
+  - Boolean literals represent the two truth values: `true` and `false`.
+
+- **Special Literals**:
+  - **Nil Literal**: The literal `nil` represents the absence of a value in Cadence’s optional types.
+  - **Resource Literals**: Resources are created and destroyed using specific syntax in Cadence, but the core value does not have a literal representation. Instead, resources are referenced through special constructs like `create` and `destroy`.
+
+- **Array Literals**:
+  - **TODO**
+
+- **Dictionary Literals**:
+  - **TODO**
+
+---
+
+### 5. Comments
+
+Comments in Cadence are non-executable sections of the code that can be used for documentation, explanation, or temporarily disabling code. Cadence supports two types of comments:
+
+- **Single-line Comments**:
+  - Single-line comments start with `//` and continue until the end of the line.
+  - Example: 
+    ```cadence
+    // This is a single-line comment
+    let value = 42  // This is also a single-line comment
+    ```
+
+- **Multi-line Comments**:
+  - Multi-line comments start with `/*` and end with `*/`. They can span multiple lines and can be nested.
+  - Example:
+    ```cadence
+    /* This is a 
+       multi-line comment */
+    ```
+
+  - **Nested Comments**:
+    - Cadence allows multi-line comments to be nested, which means you can comment out blocks of code that contain multi-line comments without causing errors.
+    - Example:
+      ```cadence
+      /* This is a multi-line comment 
+         /* This is a nested comment */ 
+         Ending the outer comment */
+      ```
+
+---
+
+This lexical structure ensures a clear and consistent foundation for writing Cadence programs, supporting both readability and maintainability across a variety of use cases within the Flow blockchain.
+
+## III. Syntax and Grammar
+
+The syntax and grammar of Cadence define the formal structure of programs written in the language. This section outlines the key elements that make up Cadence programs, including how contracts, interfaces, functions, variables, and statements are declared and used, as well as the rules governing expressions and control flow.
+
+---
+
+### 1. Program Structure
+
+A Cadence program is composed of a combination of **contracts**, **interfaces**, **transactions**, and **scripts**. These building blocks allow developers to create decentralized applications that interact with the Flow blockchain.
```

Comment:
Scripts are just a conceptual feature of the Flow blockchain, but not a dedicated language feature, so maybe remove it here and below.

In general, Cadence programs consist of zero or more declarations. However, there are three "kinds" of programs / conventions for programs:
- Contracts and contract interfaces: Only have a single contract or contract declaration at the top-level. Persistent, deployed to the blockchain
- Transactions: Only have a single transaction declaration at the top-level. Ephemeral, submitted to the blockchain to perform read/write operations
- Scripts: Only have a single function declaration at the top-level, named `main`.

Maybe merge this explanation of the structure with the purposes below

Reply:
Scripts have been removed from this.